### ISSUE TITLE
next update round: New Year's special edition

### ISSUE THREAD

**Issue by IndrajeetPatil** [easystats/easystats#92]
Scheduled for some time in January

- [x] insight 
- [x] modelbased
- [ ] bayestestR
- [x] effectsize
- [x] parameters
- [ ] performance
- [ ] correlation
- [ ] see

**Comment by IndrajeetPatil** [comment]
From the `ggstatsplot`'s next release cycle perspective (also in January, **after** `easystats` is updated), it would be nice to see the following issues resolved:

- [ ] https://github.com/easystats/insight/issues/222
- [ ] https://github.com/easystats/parameters/issues/345
- [ ] https://github.com/easystats/effectsize/issues/214 (Cohen's _d_, Hedge's _g_, Cohen's _g_ for McNemar)

**Comment by IndrajeetPatil** [comment]
`CRAN` submission portal will reopen on the 5th of January, so some time after that.

**Comment by mattansb** [comment]
(not sure effectsize 0.4.5 will be ready by then...)

**Comment by IndrajeetPatil** [comment]
No, no, this is just to say that we can't submit any time before then anyway. 

We can also start the next cycle in **February**. There is no urgent need right now to enter a new release cycle.

**Comment by DominiqueMakowski** [comment]
there are important fixes in the displays, printing etc.

At each release cycle I feel like we fixed some super duper critical issues and bugs and I wonder how could we live in peace 'til then

**Comment by strengejacke** [comment]
If MCMCglmm won't be fixed in time, there will likely be a request to update [parameters](https://cran.r-project.org/web/checks/check_results_parameters.html) and [insight](https://cran.r-project.org/web/checks/check_results_insight.html) on CRAN, so there probably won't be much more additions to the packages beyond their current state.

**Comment by IndrajeetPatil** [comment]
I see. Well, by the time the update is requested, and it is submitted, we will still have added a few more things. 

Maybe we should have these release cycles only for **major** version updates, but I am not sure if we are following any strict principles when it comes to release versions. Right now, I have only specified the next `effectsize` version.

**Comment by mattansb** [comment]
effectsize is essentially ready. Need insight and parameters updated first. No pressure (:

**Comment by strengejacke** [comment]
insight and parameters are also ready, however, need to update sjstats first, and https://github.com/MHaringa/insurancerating/issues/3#issuecomment-748512088 needs to be submitted first.

The check issues from insight on CRAN are caused by MCMCglmm, and I didn't get any emails from CRAN yet to update insight. Thus, there is also no pressure from my side, maybe we can aim at the second half of January or beginning of February for submission, depending on when the above mentioned points are solved.

**Comment by IndrajeetPatil** [comment]
As usual, my packages, all of which are going to fall apart with the new submission, are also ready for submission once the `easystats` packages are updated.

Just let `CRAN` know that you have notified the maintainer, and he will be submitting newer versions soon.

**Comment by strengejacke** [comment]
@DominiqueMakowski [parameters checks](https://cran.r-project.org/web/checks/check_results_parameters.html) are failing on CRAN, due to changes in MCMCglmm. It's likely that I'll be noted to update the package soon, so I have to submit _insight_ before.

Please tell me the current state of all those new functions you added to _insight_, anything urgent that needs to be done before submission?

**Comment by DominiqueMakowski** [comment]
afaik no, all the "critical" stuff is there already (i.e., the ellipsis tidier and the loglik)

**Comment by strengejacke** [comment]
I thought about having a dedicated method `is_nested_models()` or similar, that wrap `ellipses_info()` just to check if models are nested?

**Comment by DominiqueMakowski** [comment]
basically exporting the current internal used to check it? Yeah that would surely be useful

**Comment by IndrajeetPatil** [comment]
Okay, just got an e-mail from `CRAN` that I must submit new versions of my packages before **26.01.2021**.

Will we have submitted new versions of `easystats` packages to `CRAN` by then?

**Comment by DominiqueMakowski** [comment]
16 days easy peasy :)

**Comment by strengejacke** [comment]
If anyone has an idea why these tests fail, while locally at least the code runs, I'd be very happy...

https://github.com/easystats/insight/runs/1684731894#step:10:426

When testing, `get_residuals(model, weighted = TRUE)` returns `NULL`, running the code outside tests works fine. I'm disabling tests for now...

**Comment by DominiqueMakowski** [comment]
Here's what I see locally...

<details>
  <summary>Checks log</summary>

```
v  Comparing 'spelling.Rout' to 'spelling.Rout.save' ...
E  Running 'testthat.R' [66s] (1m 5.9s)
   Running the tests in 'tests/testthat.R' failed.
   Last 13 lines of output:
     [13] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
     [25] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
      --- function from context --- 
     function (expr, name, parentenv, handler) 
     {
         .Internal(.addCondHands(name, list(handler), parentenv, environment(), 
             FALSE))
         expr
     }
     <bytecode: 0x00000000130868f0>
     <environment: 0x0000000027e86cc0>
      --- function search by body ---
      ----------- END OF FAILURE REPORT -------------- 
     Fatal error: length > 1 in coercion to logical
     
v  checking for unstated dependencies in vignettes (375ms)
v  checking package vignettes in 'inst/doc' ... 
v  checking re-building of vignette outputs (8.9s)
   
   See
     'C:/Users/Dom/Dropbox/RECHERCHE/N/easystats/insight.Rcheck/00check.log'
   for details.
   
-- R CMD check results ----------------------------------- insight 0.11.1.1 ----
Duration: 4m 26.5s

> checking tests ...
  See below...

> checking package dependencies ... NOTE
  Packages suggested but not available for checking:
    'censReg', 'cgam', 'coxme', 'crch', 'feisr', 'fixest', 'gam', 'gbm',
    'gmnl', 'gt', 'JM', 'logistf', 'multgee', 'rms', 'robustlmm',
    'speedglm', 'tripack', 'truncreg'

-- Test failures ------------------------------------------------- testthat ----

> library(testthat)
> library(insight)
> 
> if (length(strsplit(packageDescription("insight")$Version, "\\.")[[1]]) > 3) {
+   Sys.setenv("RunAllinsightTests" = "yes")
+ } else {
+   Sys.setenv("RunAllinsightTests" = "no")
+ }
> 
> test_check("insight")
-- 1. Failure: find_formula (@test-BayesFactorBF.R#39)  ----------------------------------------------------------
find_formula(t1) not equal to list(conditional = y ~ 1).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 2. Failure: find_formula (@test-BayesFactorBF.R#40)  ----------------------------------------------------------
find_formula(t2) not equal to list(conditional = y ~ group).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 3. Failure: find_formula (@test-BayesFactorBF.R#41)  ----------------------------------------------------------
find_formula(t2d) not equal to list(conditional = y ~ 1).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 4. Failure: find_formula (@test-BayesFactorBF.R#101)  ---------------------------------------------------------
find_formula(x) not equal to list(conditional = as.formula("len ~ supp + dose + supp:dose")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 5. Failure: find_formula (@test-BayesFactorBF.R#121)  ---------------------------------------------------------
find_formula(x) not equal to list(...).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 6. Failure: find_formula (@test-BayesFactorBF.R#222)  ---------------------------------------------------------
find_formula(x) not equal to list(conditional = as.formula("len ~ supp + dose")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 7. Failure: find_formula (@test-BayesFactorBF.R#242)  ---------------------------------------------------------
find_formula(x) not equal to list(conditional = as.formula("len ~ supp + dose")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 8. Error: find_formula (@test-GLMMadaptive.R#138)  ------------------------------------------------------------
unused argument (ignore_attr = TRUE)
1: expect_identical(names(find_formula(m)), c("conditional", "random", "zero_inflated", "zero_inflated_random"), ignore_attr = TRUE) at testthat/test-GLMMadaptive.R:138

-- 9. Failure: find_formula (@test-MCMCglmm.R#70)  ---------------------------------------------------------------
find_formula(m1) not equal to list(conditional = as.formula("PO ~ plate"), random = as.formula("~FSfamily")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 10. Failure: find_formula (@test-aovlist.R#58)  ---------------------------------------------------------------
find_formula(m1) not equal to list(conditional = as.formula("yield ~ N * P * K + Error(block)")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 11. Failure: find_formula (@test-aovlist.R#64)  ---------------------------------------------------------------
find_formula(m2) not equal to list(conditional = as.formula("yield ~ N * P * K")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 12. Failure: find_formula (@test-betabin.R#61)  ---------------------------------------------------------------
find_formula(m1) not equal to list(...).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 13. Failure: find_formula (@test-betareg.R#48)  ---------------------------------------------------------------
find_formula(m1) not equal to list(conditional = as.formula("yield ~ batch + temp")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 14. Failure: find_formula (@test-betareg.R#53)  ---------------------------------------------------------------
find_formula(m2) not equal to list(conditional = as.formula("I(food/income) ~ income + persons")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 15. Failure: find_formula (@test-bigglm.R#79)  ----------------------------------------------------------------
find_formula(m1) not equal to list(conditional = as.formula("count ~ mined + log(cover) + sample")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 16. Failure: find_formula (@test-blmer.R#124)  ----------------------------------------------------------------
find_formula(m1, component = "conditional") not equal to list(conditional = as.formula("Reaction ~ Days"), random = as.formula("~1 + Days | Subject")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 17. Failure: find_formula (@test-blmer.R#132)  ----------------------------------------------------------------
find_formula(m2, component = "conditional") not equal to list(...).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

Model was fitted with uninformative (flat) priors!
-- 18. Failure: find_formula (@test-clm.R#50)  -------------------------------------------------------------------
find_formula(m1) not equal to list(conditional = as.formula("rating ~ temp * contact")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 19. Failure: find_formula (@test-clm2.R#65)  ------------------------------------------------------------------
find_formula(m1) not equal to list(conditional = as.formula("Sat ~ Infl + Type + Cont")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 20. Failure: find_formula (@test-clmm.R#93)  ------------------------------------------------------------------
find_formula(m1) not equal to list(conditional = as.formula("rating ~ temp + contact"), random = as.formula("~1 | judge")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 21. Failure: find_formula (@test-clmm.R#102)  -----------------------------------------------------------------
find_formula(m2) not equal to list(...).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 22. Failure: find_formula (@test-coxph.R#47)  -----------------------------------------------------------------
find_formula(m1) not equal to list(conditional = as.formula("Surv(time, status) ~ sex + age + ph.ecog")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 23. Failure: find_formula (@test-cpglmm.R#61)  ----------------------------------------------------------------
find_formula(m1, component = "conditional") not equal to list(conditional = as.formula("RLD ~ Stock + Spacing"), random = as.formula("~1 | Plant")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 24. Failure: find_formula (@test-crq.R#51)  -------------------------------------------------------------------
find_formula(m1) not equal to list(conditional = as.formula("survival::Surv(pmax(y, c), d, type = \"left\") ~ x")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

Gu & Wahba 4 term additive model
Gu & Wahba 4 term additive model
-- 25. Failure: find_formula (@test-gam.R#70)  -------------------------------------------------------------------
find_formula(m1) not equal to list(conditional = as.formula("y ~ s(x0) + s(x1) + s(x2) + s(x3)")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 26. Failure: find_formula (@test-gam.R#76)  -------------------------------------------------------------------
find_formula(m2) not equal to list(conditional = as.formula("y ~ s(x2) + s(x3)"), zero_inflated = as.formula("~s(x0) + s(x1)")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 27. Failure: find_formula (@test-gam.R#85)  -------------------------------------------------------------------
find_formula(m3) not equal to structure(...).
Attributes: < Names: 1 string mismatch >
Attributes: < Length mismatch: comparison on first 1 components >
Attributes: < Component 1: Lengths (2, 1) differ (string compare on first 1) >
Attributes: < Component 1: 1 string mismatch >

GAMLSS-RS iteration 1: Global Deviance = 4771.925 
GAMLSS-CG iteration 1: Global Deviance = 4771.013 
GAMLSS-CG iteration 2: Global Deviance = 4770.994 
GAMLSS-CG iteration 3: Global Deviance = 4770.994 
-- 28. Failure: find_formula (@test-gamlss.R#51)  ----------------------------------------------------------------
find_formula(m1) not equal to list(...).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

4 term additive + random effectGu & Wahba 4 term additive model

 Maximum number of PQL iterations:  20 
-- 29. Failure: find_formula (@test-gamm.R#57)  ------------------------------------------------------------------
find_formula(m1) not equal to list(conditional = as.formula("y ~ s(x0) + s(x1) + s(x2)"), random = as.formula("~1 | fac")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

Gu & Wahba 4 term additive model
4 term additive + random effectGu & Wahba 4 term additive model

 Maximum number of PQL iterations:  20 

 Maximum number of PQL iterations:  20 
-- 30. Failure: find_formula-gamm-1 (@test-gamm.R#157)  ----------------------------------------------------------
find_formula(m1) not equal to list(...).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 31. Failure: find_formula-gamm-2 (@test-gamm.R#166)  ----------------------------------------------------------
find_formula(m2) not equal to list(conditional = as.formula("y ~ s(x0) + s(x1) + s(x2)")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 32. Failure: find_formula-gamm-3 (@test-gamm.R#174)  ----------------------------------------------------------
find_formula(m3) not equal to list(conditional = as.formula("y ~ s(x0) + s(x1) + s(x2)"), random = as.formula("~1 | g")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

Gu & Wahba 4 term additive model
-- 33. Failure: find_formula (@test-gamm4.R#63)  -----------------------------------------------------------------
find_formula(m1) not equal to list(conditional = as.formula("y ~ s(x0) + x1 + s(x2)"), random = as.formula("~1 | fac")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

(Intercept)    tensionM    tensionH 
   36.38889   -10.00000   -14.72222 
-- 34. Failure: find_formula (@test-gee.R#55)  -------------------------------------------------------------------
find_formula(m1) not equal to list(conditional = as.formula("breaks ~ tension"), random = as.formula("~wool")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >

-- 35. Failure: find_formula (@test-geeglm.R#63)  ----------------------------------------------------------------
find_formula(m1) not equal to list(conditional = as.formula("breaks ~ tension"), random = as.formula("~wool")).
Attributes: < names for target but not for current >
Attributes: < Length mismatch: comparison on first 0 components >


SAMPLING FOR MODEL 'continuous' NOW (CHAIN 1).
Chain 1: 
Chain 1: Gradient evaluation took 0 seconds
Chain 1: 1000 transitions using 10 leapfrog steps per transition would take 0 seconds.
Chain 1: Adjust your expectations accordingly!
Chain 1: 
Chain 1: 
Chain 1: Iteration:    1 / 2000 [  0%]  (Warmup)
Chain 1: Iteration:  200 / 2000 [ 10%]  (Warmup)
Chain 1: Iteration:  400 / 2000 [ 20%]  (Warmup)
Chain 1: Iteration:  600 / 2000 [ 30%]  (Warmup)
Chain 1: Iteration:  800 / 2000 [ 40%]  (Warmup)
Chain 1: Iteration: 1000 / 2000 [ 50%]  (Warmup)
Chain 1: Iteration: 1001 / 2000 [ 50%]  (Sampling)
Chain 1: Iteration: 1200 / 2000 [ 60%]  (Sampling)
Chain 1: Iteration: 1400 / 2000 [ 70%]  (Sampling)
Chain 1: Iteration: 1600 / 2000 [ 80%]  (Sampling)
Chain 1: Iteration: 1800 / 2000 [ 90%]  (Sampling)
Chain 1: Iteration: 2000 / 2000 [100%]  (Sampling)
Chain 1: 
Chain 1:  Elapsed Time: 0.048 seconds (Warm-up)
Chain 1:                0.063 seconds (Sampling)
Chain 1:                0.111 seconds (Total)
Chain 1: 

SAMPLING FOR MODEL 'continuous' NOW (CHAIN 2).
Chain 2: 
Chain 2: Gradient evaluation took 0 seconds
Chain 2: 1000 transitions using 10 leapfrog steps per transition would take 0 seconds.
Chain 2: Adjust your expectations accordingly!
Chain 2: 
Chain 2: 
Chain 2: Iteration:    1 / 2000 [  0%]  (Warmup)
Chain 2: Iteration:  200 / 2000 [ 10%]  (Warmup)
Chain 2: Iteration:  400 / 2000 [ 20%]  (Warmup)
Chain 2: Iteration:  600 / 2000 [ 30%]  (Warmup)
Chain 2: Iteration:  800 / 2000 [ 40%]  (Warmup)
Chain 2: Iteration: 1000 / 2000 [ 50%]  (Warmup)
Chain 2: Iteration: 1001 / 2000 [ 50%]  (Sampling)
Chain 2: Iteration: 1200 / 2000 [ 60%]  (Sampling)
Chain 2: Iteration: 1400 / 2000 [ 70%]  (Sampling)
Chain 2: Iteration: 1600 / 2000 [ 80%]  (Sampling)
Chain 2: Iteration: 1800 / 2000 [ 90%]  (Sampling)
Chain 2: Iteration: 2000 / 2000 [100%]  (Sampling)
Chain 2: 
Chain 2:  Elapsed Time: 0.045 seconds (Warm-up)
Chain 2:                0.062 seconds (Sampling)
Chain 2:                0.107 seconds (Total)
Chain 2: 

SAMPLING FOR MODEL 'continuous' NOW (CHAIN 3).
Chain 3: 
Chain 3: Gradient evaluation took 0 seconds
Chain 3: 1000 transitions using 10 leapfrog steps per transition would take 0 seconds.
Chain 3: Adjust your expectations accordingly!
Chain 3: 
Chain 3: 
Chain 3: Iteration:    1 / 2000 [  0%]  (Warmup)
Chain 3: Iteration:  200 / 2000 [ 10%]  (Warmup)
Chain 3: Iteration:  400 / 2000 [ 20%]  (Warmup)
Chain 3: Iteration:  600 / 2000 [ 30%]  (Warmup)
Chain 3: Iteration:  800 / 2000 [ 40%]  (Warmup)
Chain 3: Iteration: 1000 / 2000 [ 50%]  (Warmup)
Chain 3: Iteration: 1001 / 2000 [ 50%]  (Sampling)
Chain 3: Iteration: 1200 / 2000 [ 60%]  (Sampling)
Chain 3: Iteration: 1400 / 2000 [ 70%]  (Sampling)
Chain 3: Iteration: 1600 / 2000 [ 80%]  (Sampling)
Chain 3: Iteration: 1800 / 2000 [ 90%]  (Sampling)
Chain 3: Iteration: 2000 / 2000 [100%]  (Sampling)
Chain 3: 
Chain 3:  Elapsed Time: 0.049 seconds (Warm-up)
Chain 3:                0.065 seconds (Sampling)
Chain 3:                0.114 seconds (Total)
Chain 3: 

SAMPLING FOR MODEL 'continuous' NOW (CHAIN 4).
Chain 4: 
Chain 4: Gradient evaluation took 0 seconds
Chain 4: 1000 transitions using 10 leapfrog steps per transition would take 0 seconds.
Chain 4: Adjust your expectations accordingly!
Chain 4: 
Chain 4: 
Chain 4: Iteration:    1 / 2000 [  0%]  (Warmup)
Chain 4: Iteration:  200 / 2000 [ 10%]  (Warmup)
Chain 4: Iteration:  400 / 2000 [ 20%]  (Warmup)
Chain 4: Iteration:  600 / 2000 [ 30%]  (Warmup)
Chain 4: Iteration:  800 / 2000 [ 40%]  (Warmup)
Chain 4: Iteration: 1000 / 2000 [ 50%]  (Warmup)
Chain 4: Iteration: 1001 / 2000 [ 50%]  (Sampling)
Chain 4: Iteration: 1200 / 2000 [ 60%]  (Sampling)
Chain 4: Iteration: 1400 / 2000 [ 70%]  (Sampling)
Chain 4: Iteration: 1600 / 2000 [ 80%]  (Sampling)
Chain 4: Iteration: 1800 / 2000 [ 90%]  (Sampling)
Chain 4: Iteration: 2000 / 2000 [100%]  (Sampling)
Chain 4: 
Chain 4:  Elapsed Time: 0.049 seconds (Warm-up)
Chain 4:                0.065 seconds (Sampling)
Chain 4:                0.114 seconds (Total)
Chain 4: 
 ----------- FAILURE REPORT -------------- 
 --- failure: length > 1 in coercion to logical ---
 --- srcref --- 
: 
 --- package (from environment) --- 
insight
 --- call from context --- 
doTryCatch(return(expr), name, parentenv, handler)
 --- call from argument --- 
!is.null(res_resp) && res_resp == res_dev
 --- R stacktrace ---
where 1: doTryCatch(return(expr), name, parentenv, handler)
where 2: tryCatchOne(expr, names, parentenv, handlers[[1L]])
where 3: tryCatchList(expr, classes, parentenv, handlers)
where 4: tryCatch({
    res_resp <- as.vector(get_residuals(x, weighted = FALSE, 
        type = "response"))
    res_dev <- as.vector(get_residuals(x, weighted = FALSE, type = "deviance"))
    if (!is.null(w) && !is.null(res_dev)) {
        if (!is.null(res_resp) && res_resp == res_dev) {
            res_dev <- res_dev * w^0.5
        }
        res_dev <- res_dev[!is.na(w) & w != 0]
    }
    else if (verbose) {
        if (is.null(w)) {
            warning("Can't calculate weighted residuals from model. Model doesn't seem to have weights.", 
                call. = FALSE)
        }
        else if (is.null(res_dev)) {
            warning("Can't calculate weighted residuals from model. Could not extract deviance-residuals.", 
                call. = FALSE)
        }
    }
    res_dev
}, error = function(e) {
    if (verbose) {
        warning("Can't calculate weighted residuals from model.", 
            call. = FALSE)
    }
    warning(e$message, call. = TRUE)
    NULL
})
where 5: .weighted_residuals(x, verbose)
where 6: get_residuals.default(m, weighted = TRUE)
where 7: insight::get_residuals(m, weighted = TRUE)
where 8: eval_bare(get_expr(quo), get_env(quo))
where 9: quasi_label(enquo(object), label, arg = "object")
where 10 at testthat/test-get_residuals.R#29: expect_equal(insight::get_residuals(m, weighted = TRUE), as.vector(weighted.residuals(m)))
where 11: eval(code, test_env)
where 12: eval(code, test_env)
where 13: withCallingHandlers({
    eval(code, test_env)
    if (!handled && !is.null(test)) {
        skip_empty()
    }
}, expectation = handle_expectation, skip = handle_skip, warning = handle_warning, 
    message = handle_message, error = handle_error)
where 14: doTryCatch(return(expr), name, parentenv, handler)
where 15: tryCatchOne(expr, names, parentenv, handlers[[1L]])
where 16: tryCatchList(expr, names[-nh], parentenv, handlers[-nh])
where 17: doTryCatch(return(expr), name, parentenv, handler)
where 18: tryCatchOne(tryCatchList(expr, names[-nh], parentenv, handlers[-nh]), 
    names[nh], parentenv, handlers[[nh]])
where 19: tryCatchList(expr, classes, parentenv, handlers)
where 20: tryCatch(withCallingHandlers({
    eval(code, test_env)
    if (!handled && !is.null(test)) {
        skip_empty()
    }
}, expectation = handle_expectation, skip = handle_skip, warning = handle_warning, 
    message = handle_message, error = handle_error), error = handle_fatal, 
    skip = function(e) {
    })
where 21: test_code(desc, code, env = parent.frame())
where 22 at testthat/test-get_residuals.R#10: test_that("get_residuals - lm", {
    m <- lm(am ~ cyl, weights = w, data = mtcars)
    expect_equal(as.vector(get_residuals(m, weighted = FALSE)), 
        as.vector(residuals(m)))
    expect_equal(as.vector(get_residuals(m, weighted = FALSE, 
        type = "deviance")), as.vector(residuals(m, type = "deviance")))
    expect_equal(get_weights(m), weights(m))
    expect_equal(as.vector(insight::get_residuals(m)), as.vector(residuals(m)))
    expect_equal(insight::get_residuals(m, weighted = TRUE), 
        as.vector(weighted.residuals(m)))
})
where 23: eval(code, test_env)
where 24: eval(code, test_env)
where 25: withCallingHandlers({
    eval(code, test_env)
    if (!handled && !is.null(test)) {
        skip_empty()
    }
}, expectation = handle_expectation, skip = handle_skip, warning = handle_warning, 
    message = handle_message, error = handle_error)
where 26: doTryCatch(return(expr), name, parentenv, handler)
where 27: tryCatchOne(expr, names, parentenv, handlers[[1L]])
where 28: tryCatchList(expr, names[-nh], parentenv, handlers[-nh])
where 29: doTryCatch(return(expr), name, parentenv, handler)
where 30: tryCatchOne(tryCatchList(expr, names[-nh], parentenv, handlers[-nh]), 
    names[nh], parentenv, handlers[[nh]])
where 31: tryCatchList(expr, classes, parentenv, handlers)
where 32: tryCatch(withCallingHandlers({
    eval(code, test_env)
    if (!handled && !is.null(test)) {
        skip_empty()
    }
}, expectation = handle_expectation, skip = handle_skip, warning = handle_warning, 
    message = handle_message, error = handle_error), error = handle_fatal, 
    skip = function(e) {
    })
where 33: test_code(NULL, exprs, env)
where 34: source_file(path, new.env(parent = env), chdir = TRUE, wrap = wrap)
where 35: force(code)
where 36: with_reporter(reporter = reporter, start_end_reporter = start_end_reporter, 
    {
        reporter$start_file(basename(path))
        lister$start_file(basename(path))
        source_file(path, new.env(parent = env), chdir = TRUE, 
            wrap = wrap)
        reporter$end_context()
        reporter$end_file()
    })
where 37: FUN(X[[i]], ...)
where 38: lapply(paths, test_file, env = env, reporter = current_reporter, 
    start_end_reporter = FALSE, load_helpers = FALSE, wrap = wrap)
where 39: force(code)
where 40: with_reporter(reporter = current_reporter, results <- lapply(paths, 
    test_file, env = env, reporter = current_reporter, start_end_reporter = FALSE, 
    load_helpers = FALSE, wrap = wrap))
where 41: test_files(paths, reporter = reporter, env = env, stop_on_failure = stop_on_failure, 
    stop_on_warning = stop_on_warning, wrap = wrap)
where 42: test_dir(path = test_path, reporter = reporter, env = env, filter = filter, 
    ..., stop_on_failure = stop_on_failure, stop_on_warning = stop_on_warning, 
    wrap = wrap)
where 43: test_package_dir(package = package, test_path = test_path, filter = filter, 
    reporter = reporter, ..., stop_on_failure = stop_on_failure, 
    stop_on_warning = stop_on_warning, wrap = wrap)
where 44: test_check("insight")

 --- value of length: 32 type: logical ---
 [1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[13] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[25] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 --- function from context --- 
function (expr, name, parentenv, handler) 
{
    .Internal(.addCondHands(name, list(handler), parentenv, environment(), 
        FALSE))
    expr
}
<bytecode: 0x00000000130868f0>
<environment: 0x0000000027e86cc0>
 --- function search by body ---
 ----------- END OF FAILURE REPORT -------------- 
Fatal error: length > 1 in coercion to logical


1 error x | 0 warnings v | 1 note x
Error: R CMD check found ERRORs
Execution halted

Exited with status 1.
```


</details>

**Comment by strengejacke** [comment]
> find_formula(t1) not equal to list(conditional = y ~ 1).

Ok, that's weird, looks like your packages are not up to date?


 --- failure: length > 1 in coercion to logical ---
 --- srcref --- 
: 
 --- package (from environment) --- 
insight
 --- call from context --- 
doTryCatch(return(expr), name, parentenv, handler)
 --- call from argument --- 
!is.null(res_resp) && res_resp == res_dev


That could be the reason!

**Comment by strengejacke** [comment]
Ok, running final checks on insight and will then submit first package. I also got a msg from CRAN regarding _parameters_.

**Comment by strengejacke** [comment]
@DominiqueMakowski Failing test resolved! I wonder why I didn't get that output from local tests?!? Anyway, works now. What about the _psycho_ package? That *may* break after submitting insight (though I guess it actually should fail after submitting parameters, not insight). Can you prepare an update?

**Comment by DominiqueMakowski** [comment]
Yup should be quite fast, I'll submit the updated right after insight is up (basically reoriginating all format_p)

**Comment by mattansb** [comment]
`effectsize` is ready - will upload when `insight` and `parameters` are updated.

**Comment by strengejacke** [comment]
Ok, will submit insight in the course of this day (earliest in ~4 hours). Anything important missing?

**Comment by DominiqueMakowski** [comment]
Our version of estfun() so we can at least remove sandwich from performances deps?

**Comment by strengejacke** [comment]
> Our version of estfun() so we can at least remove sandwich from performances deps?

I'd suggest we first have this as an internal function in _performance_, so we have some more time for proper implementation.

**Comment by DominiqueMakowski** [comment]
agreed

**Comment by strengejacke** [comment]
> `effectsize` is ready - will upload when `insight` and `parameters` are updated.

@mattansb can you check the latest effectsize with the latest parameters (main branch), to ensure all checks pass, before I submit to CRAN?

**Comment by mattansb** [comment]
Gh action should be testing with the main branch. But will run a manual test tomorrow morning 👍

**Comment by strengejacke** [comment]
@DominiqueMakowski we need to submit an update for modelbased after insight was submitted. I think I fixed all the affected tests.

**Comment by mattansb** [comment]
Is insight on CRAN?

**Comment by strengejacke** [comment]
Not yet, revdep checks were complaining about failing tests in modelbased. I would say that this is expected and the maintainer is informed.

**Comment by mattansb** [comment]
😂

**Comment by strengejacke** [comment]
> Thanks, on its way to CRAN.
>
> Team, this breaks modelbased.
>
> Best,

@DominiqueMakowski please submit an update of modelbased, so we can continue with parameters.

**Comment by DominiqueMakowski** [comment]
https://github.com/easystats/modelbased/actions/runs/486932457 There are still some issues related it seems to stan on some systems, but we'll see. I'll run the winchecks now

**Comment by DominiqueMakowski** [comment]
All winbuild checks are OK, so will submit to CRAN now

**Comment by DominiqueMakowski** [comment]
package modelbased_0.5.0.tar.gz is on its way to CRAN.

**Comment by IndrajeetPatil** [comment]
@DominiqueMakowski The link to website (https://easystats.github.io/bayestestR/) is missing from `bayestestR`'s `DESCRIPTION` file. Can you please add it?

**Comment by DominiqueMakowski** [comment]
What about bayestestR should we submit now or no pressure?

**Comment by IndrajeetPatil** [comment]
I'd vote for pushing `bayestestR` to `CRAN` as well, maybe the version bump can be minor instead of major, up to you. 
But it'd be good to keep the entire ecosystem in sync with each other on `CRAN`.

**Comment by strengejacke** [comment]
_parameters_ is on CRAN. @DominiqueMakowski you need to submit an update for psycho, @IndrajeetPatil statsExpressions for you to submit. :-)

**Comment by mattansb** [comment]
Should I wait for `bayestestR` now? Or go ahead and submit `effectsize`?

**Comment by strengejacke** [comment]
If you don't need the bayestestR update, go ahead.

**Comment by mattansb** [comment]
Going ahead then.

**Comment by IndrajeetPatil** [comment]
Will `performance` be updated after `bayestestR`? I will need it for my `tidyBF` update.

**Comment by strengejacke** [comment]
is there a specific date you need to submit? _performance_ may take some further days until all the test-stuff is implemented.

**Comment by IndrajeetPatil** [comment]
I will need it asap because my entire ecosystem is currently broken and the increasing number of users will be exposed to these breakages the longer I wait to update them.

**Comment by strengejacke** [comment]
@DominiqueMakowski You're mainly working on _performance_ currently, maybe you can suggest a date for submission?

**Comment by strengejacke** [comment]
@IndrajeetPatil I'm wondering what new features in _performance_ you're actually relying on?

**Comment by mattansb** [comment]
One `effectsize 0.4.3` is on CRAN (should happen soon, all checks pass), `correlation` and `psycho` need to be fixed and updated, due to changes in `adjust()` (which really should be moved to `correlation`...). `statsExpressions` is also affected.

**Comment by strengejacke** [comment]
> which really should be moved to correlation...

Do you think that's a good idea? correlation also depends on `effectsize::ranktransform()`, so we would add another circular dependency...

**Comment by mattansb** [comment]
It won't because `adjust` isn't used by `effectsize` at all (:

**Comment by DominiqueMakowski** [comment]
Regarding performance, most of the feature-breaking stuff is done, now only minor things are eventually remaining, such as getting rid of one of the dependency (i.e., reimplement and extend .get_gradients) and such, so it's mostly your call Daniel when exactly to submit but I'd say it beginning of next week is reasonable

**Comment by IndrajeetPatil** [comment]
> I'm wondering what new features in performance you're actually relying on?

@strengejacke Turns out, I am not using anything new. So no need to update `performance` asap. Sorry for the confusion 😐 

So the only thing I need to wait for now is `effectsize` to make it to `CRAN` then.

**Comment by DominiqueMakowski** [comment]
> which really should be moved to correlation...

Stahp wanting to get rid of all my functions ^^ 

It was initially put in effectsize because its in the same vein as standardize and normalize, but in reality all of these would be better located ***in our infamous data related package***

**Comment by mattansb** [comment]
Not get rid, just have it all nice and tidy - with everything in its place 😊

**Comment by strengejacke** [comment]
> in our infamous potential data related package

True, so we should not change anything. That package could probably something which maybe only depends on _insight_?

**Comment by mattansb** [comment]
Yes, prob

**Comment by IndrajeetPatil** [comment]
Maybe we can create another issue here to keep a list of all data-cleaning related functions that can be part of this new package?

**Comment by strengejacke** [comment]
@mattansb if you don't have submitted effectsize to CRAN yet, wait a minute... not sure if there's a bug in `adjust()`?

**Comment by mattansb** [comment]
Already did... 🥴

**Comment by strengejacke** [comment]
Ok, and it will pass all checks, so you don't need to resubmit?

**Comment by mattansb** [comment]
Yup.... It already did, and I was asked to address reverse dependencies, and did... Just waiting to get "on its way"

**Comment by strengejacke** [comment]
@DominiqueMakowski do we need to update _performance_ and _bayestestR_ in this update round? Or could we skip these two packages and have some more times for development?

**Comment by mattansb** [comment]
`effectsize` is on CRAN

**Comment by strengejacke** [comment]
Is it the "working" version with the fixed bug in `adjust()`?

**Comment by mattansb** [comment]
Yes!!

**Comment by DominiqueMakowski** [comment]
performance needs bayestestR, and there's no reason not to submit bayestestR... I'll do that asap

**Comment by DominiqueMakowski** [comment]
package bayestestR_0.8.1.tar.gz does not pass the incoming checks automatically, please see the following pre-tests:
Windows: <https://win-builder.r-project.org/incoming_pretest/bayestestR_0.8.1_20210119_051034/Windows/00check.log>
Status: 1 WARNING, 1 NOTE
Debian: <https://win-builder.r-project.org/incoming_pretest/bayestestR_0.8.1_20210119_051034/Debian/00check.log>
Status: 1 NOTE

**Comment by mattansb** [comment]
Where do we use `stringi`? 🤔

**Comment by DominiqueMakowski** [comment]
Pls ignore the Windows error in the report you hgot, but pls fix:


   Unknown, possibly mis-spelled, fields in DESCRIPTION:
     'Remotes'

Please fix and resubmit.

**Comment by strengejacke** [comment]
> performance needs bayestestR, and there's no reason not to submit bayestestR... I'll do that asap

I don't see anything (https://easystats.github.io/bayestestR/news/index.html) that's important for _performance_?

**Comment by DominiqueMakowski** [comment]
bf_models dealing with non ellipsis inputs 😬 (as required in test_perf)

**Comment by strengejacke** [comment]
Ah, ok. You could _maybe_, _probably_ update the news file... 😬

**Comment by strengejacke** [comment]
> One `effectsize 0.4.3` is on CRAN (should happen soon, all checks pass), `correlation` and `psycho` need to be fixed and updated, due to changes in `adjust()` (which really should be moved to `correlation`...). `statsExpressions` is also affected.

I think _correlation_ is no longer affected, since the rev-dep issue was due to the buggy `adjust()`, which was fixed before submission. Checks are also okay atm. (https://cran.r-project.org/web/checks/check_results_correlation.html)

**Comment by DominiqueMakowski** [comment]
package bayestestR_0.8.1.tar.gz does not pass the incoming checks automatically, please see the following pre-tests:
Windows: <https://win-builder.r-project.org/incoming_pretest/bayestestR_0.8.1_20210120_011056/Windows/00check.log>
Status: OK
Debian: <https://win-builder.r-project.org/incoming_pretest/bayestestR_0.8.1_20210120_011056/Debian/00check.log>
Status: 1 NOTE

Thanks, we see:


* checking re-building of vignette outputs ... [472s] OK

This is long and causes the overall check time to be > 10 min.

Can this be reduced by using
  - small toy data only
  - few iterations
  - or providing precomputed results for the lengthy compuations?

Please fix and resubmit.

**Comment by DominiqueMakowski** [comment]
I think the NOTE is a false positive, it's mostly the check time last time they already made a remark 😕

**Comment by strengejacke** [comment]
Can we skip some more tests on CRAN? Or maybe disable (i.e. `dontrun{}`) examples?

**Comment by DominiqueMakowski** [comment]
yeah that's what I'm looking at, Is there a way with which we can "sort" the running time of all the examples/tests? To focus on the longest

**Comment by strengejacke** [comment]
Not sure. The local R CMD check creates a "timings" file, maybe there you can find some information on duration of tests/examples?

**Comment by DominiqueMakowski** [comment]
```
name	user	system	elapsed
area_under_curve	0	0	0	
bayesfactor	0.05	0.03	0.08	
bayesfactor_inclusion	0.01	0.00	0.01	
bayesfactor_models	0.03	0.00	0.03	
bayesfactor_parameters	0.05	0.00	0.05	
bayesfactor_restricted	0	0	0	
check_prior	0	0	0	
ci	0.03	0.00	0.03	
contr.bayes	0	0	0	
convert_bayesian_as_frequentist	6.72	0.39	7.15	
density_at	0	0	0	
describe_posterior	0.25	0.00	0.25	
describe_prior	0	0	0	
diagnostic_posterior	0	0	0	
distribution	0.03	0.00	0.03	
effective_sample	0	0	0	
equivalence_test	0.03	0.00	0.03	
estimate_density	0.19	0.00	0.19	
eti	0.03	0.00	0.03	
hdi	0.03	0.00	0.04	
map_estimate	0	0	0	
mcse	0	0	0	
mediation	0	0	0	
mhdior	0	0	0	
overlap	0.02	0.00	0.03	
p_direction	0.01	0.00	0.01	
p_map	9.17	0.00	9.17	
p_rope	0.01	0.00	0.01	
p_significance	0	0	0	
pd_to_p	0	0	0	
point_estimate	0.05	0.00	0.05	
reshape_ci	0.03	0.00	0.03	
rope	0.01	0.00	0.02	
rope_range	0	0	0	
sensitivity_to_prior	0	0	0	
sexit	0	0	0	
sexit_thresholds	0	0	0	
si	0.04	0.00	0.04	
simulate_correlation	0.07	0.00	0.66	
simulate_prior	0	0	0	
weighted_posteriors	4.35	0.13	4.62	
```

**Comment by mattansb** [comment]
Another reason to drop convert_bayesian_as_frequentist ;)

The examples in weighted posterior can be skipped.

**Comment by DominiqueMakowski** [comment]
aren't you guys supposed to be sleeping 😅

**Comment by DominiqueMakowski** [comment]
The problem is that most of the above examples are already dontrun'ed or donttest'ed or skiponcran'ed 🤔 I'm not sure which ones are the ones left that take long

**Comment by mattansb** [comment]
> aren't you guys supposed to be sleeping 😅

Sleeping? That's that thing people without infants do, no? I think I remember...

**Comment by IndrajeetPatil** [comment]
Closing this in favor of https://github.com/easystats/easystats/issues/98, since all submission henceforth will be `R >= 3.4`.

**Comment by strengejacke** [comment]
> Sleeping? That's that thing people without infants do, no? I think I remember...

I feel you, bro. Wait another 5 years, and realize you're amount of sleeping did not and still does not exceed 4 hours (unless you sleep at work).

**Comment by mattansb** [comment]
How much coffee is too much coffee?
(Trick question, obviously!)

**Comment by strengejacke** [comment]
@DominiqueMakowski what is the state of bayestestR and modelbased submission?

**Comment by strengejacke** [comment]
_bump_
